### PR TITLE
feat: waiting for dependencies to be installed in wizard   

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes UNIFY-967

### User facing changelog

Screen Shot 2022-01-28 at 2 35 53 PM

https://user-images.githubusercontent.com/5592465/151632337-e16d4ff6-af5a-4b07-917a-d7dd6fd90e5a.mov

### PR Tasks

- [x] Have tests been added/updated?

### CODE REVIEW

File: packages/launchpad/src/setup/ManualInstall.vue [hunk 1]
```diff
@@ -54,16 +67,17 @@ const projectFolder = computed(() => props.gql.currentProject?.title ?? '')
...
 const props = defineProps<{
   gql: ManualInstallFragment
+  packagesInstalled: string[]
 }>()
 
-defineEmits<{
-  (event: 'back'): void
-}>()
+const toInstall = computed(() => {
+  return props.gql.wizard.packagesToInstall
```
Comment (lmiller1990) [inline packages/launchpad/src/setup/ManualInstall.vue:74]: Do we really need this additional computed property? It's only used in one place - isn't this effectively a temp variable (that also adds another thing for the reactivity system to track?

**Review by lmiller1990 — COMMENTED** [review]
Code seems fine - I didn't try locally, but I see some potential edge cases around using `require.resolve` like this. What about the case of a monorepo? It's possible they install the module at top level, and we are polling on the wrong `require.resolve`. Or, it gets hoisted and we don't poll the right path.

I like this feature but I'm concerned about reliability. 

- If they get stuck waiting, even if they already installed, that's not ideal. Do you see any chance of an edge where this happens? 
- Another case is they just used the migration tool and already have the deps installed; maybe we should have "continue anyway; I've installed it already" button, as well.
- What about managers like pnpm and yarn2, where there's no local `node_modules`? This will never return true, since we cannot resolve the module with `require.resolve` like this.

What do you think? I see a lot of edge cases; I think we need a "proceed anyway" style button, at least.

File: packages/launchpad/src/setup/ManualInstall.vue (hunk 1 above)
Comment (elevatebart) [reply packages/launchpad/src/setup/ManualInstall.vue:74]: Yeah you are right. I will remove it.

Comment (elevatebart) [reply packages/launchpad/src/setup/ManualInstall.vue]: Yeah you are right. I will remove it.

**Comment by lmiller1990** [comment]
@elevatebart will this work for the use case of a nested repo? Eg will it detect the deps were installed a higher level? How about things like pnpm and yarn where the deps are not in the local `node_modules` at all, but in some location far away from the project?

**Comment by elevatebart** [comment]
@lmiller1990 that is indeed a good point to make.

Nested projects work, this is the reason why I cannot do proper e2e testing here. When doing `require.resolve()` node will first hit the `node_modules` directory near `package.json` then looks on directory up recursively for a `node_modules` near a `package.json`.

This process is described in the [node documentation](https://nodejs.org/api/modules.html#loading-from-node_modules-folders)

WIth `pnpm` I tested and it works. Obviously `yarn` & `npm` I have tested as well.
Regarding `yarn` v2/3 I have not tested it. I would hope it still works since we are hardly the only package that relies on this.

**Review by lmiller1990 — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Comment by baus** [comment]
@elevatebart 
> If people are blocked they can always close cypress and restart.

How likely is this to happen?

**Comment by elevatebart** [comment]
@baus From what I tested with `npm`, `pnpm` and `yarn` very very unlikely. 
If `node` cannot resolve those package, there is good chance that cypress won't be either.

The risk is really reasonnable.

File: packages/data-context/src/sources/WizardDataSource.ts [hunk 2]
```diff
@@ -37,6 +42,41 @@ export class WizardDataSource {
...
+    }
+
+    debug('packages to install: %O', packagesInitial)
+
+    const installedPackages: (string|null)[] = packagesInitial.map((p) => {
+      if (this.ctx.currentProject) {
+        debug('package checked: %s', p.package)
+
+        // At startup, node will only resolve the main files of packages it knows of.
+        // Adding a package after the app started will not be resolved in the same way
```
Comment (elevatebart) [inline packages/data-context/src/sources/WizardDataSource.ts:59]:
```suggestion
        // Adding a package after the app started will not be resolved in the same way.
```

**Review by marktnoonan — APPROVED** [review]

**Pull request merged**